libs/barista-components/breadcrumbs/src/breadcrumbs.ts

```diff
@@ -49,38 +98,394 @@ export const _DtBreadcrumbMixinBase = mixinColor<
   styleUrls: ['breadcrumbs.scss'],
   host: {
     class: 'dt-breadcrumbs',
+    '[attr.aria-haspopup]': 'this._hasHiddenItems',
   },
   inputs: ['color'],
   preserveWhitespaces: false,
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.Emulated,
 })
-export class DtBreadcrumbs extends _DtBreadcrumbMixinBase
-  implements AfterContentInit, OnDestroy {
+export class DtBreadcrumbs
+  extends _DtBreadcrumbMixinBase
+  implements OnInit, AfterContentInit, AfterViewInit, OnDestroy {
   @ContentChildren(DtBreadcrumbsItem2) private _items: QueryList<
     DtBreadcrumbsItem2
   >;
 
+  _transplantedItemsString = ``;
+
+  private _containerSizeObserver: any;
+
   private _destroy$ = new Subject<void>();
 
-  constructor(elementRef: ElementRef) {
+  private _containerContentRect$ = new Subject<DOMRect>();
+
+  private _overlayTriggerKeydown$ = new Subject<KeyboardEvent>();
+
+  private _overlayRef: OverlayRef | null = null;
+
+  private _itemsPortalsMap = new Map<DtBreadcrumbsItem2, PortalOutlet>();
+
+  private _itemsWidthMap = new Map<DtBreadcrumbsItem2, number>();
+
+  private _collapsedContainer: HTMLDivElement | null = null;
+
+  get _hasHiddenItems(): boolean {
+    return this._items.length > 0 && this._itemsPortalsMap.size > 0;
+  }
+
+  @ViewChild('collapseTrigger', { static: false }) _trigger: ElementRef | null;
+
+  constructor(
+    public elementRef: ElementRef,
+    private _componentFactoryResolver: ComponentFactoryResolver,
+    private _appRef: ApplicationRef,
+    private _injector: Injector,
+    @Inject(DOCUMENT) private _document: any,
+    private _overlay: Overlay,
+    private _changeDetectorRef: ChangeDetectorRef,
+    private _zone: NgZone,
+    private _focusMonitor: FocusMonitor,
```

Comment:
Adding all these new parameters to the constructor is a breaking change - we need to annotate the commit as such :)

Reply:
You can do this by adding BREAKING-CHANGE to the commit body when you rebase.